geode-management/src/main/java/org/apache/geode/management/configuration/Pdx.java

```diff
@@ -31,8 +32,8 @@
 
   private Boolean readSerialized;
   private ClassName pdxSerializer;
+  private AutoSerializer autoSerializer;
```

Comment:
I believe we don't need to save this autoSerializer, all it does is to set a default pdx serializer, it can just directly modify the pdxSerializer variable. But we do need am internal flag indicating that setAutoSerializer method is called, and when user call setPdxSerializer, we should check that flag and report error if that's true. This would get rid of autoserilizaer conversion/validation code.

Reply:
I think we do need this make it easy to configure a ReflectionBasedAutoSerializer from JSON. If all we have is a "pdxSerializer" then in JSON the user need to specify the ReflectionBasedAutoSerializer as a ClassName. See the initial comment on this PR for a "before and after" of what the JSON looks like for this. I think I may be missing something so lets discuss this further.

Reply:
yes, you can still do that with this. I can pair with you

Reply:
Since we want to show back the user exactly the way user configures it, then we still need the converter. Darrel is going to move the validation of mutual exclusion to the configuration objects